### DISCUSSION TITLE
Any specific configuration to make the video call work well in odoo 16 in production mode?

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/doodba-copier-template#477] [Q&A] [answered]
How's everybody?
I'm trying the wonders of the new odoo 16 and the video calling system it has, but I'm running into yellow triangles and very big latency problems. 
Poking a little on the internet, there is talk of changing the memory and processor configuration in odoo.conf, others say that it is due to the socket configuration in the reverse proxy and others talk about not exposing the database list. Another option they also say is to use an ICE server which I assume must require an extra service.

What experience have you had with this and what configuration options using the copier have been successful for you?

**Comment by pleirb** [comment]
> How's everybody? I'm trying the wonders of the new odoo 16 and the video calling system it has, but I'm running into yellow triangles and very big latency problems. Poking a little on the internet, there is talk of changing the memory and processor configuration in odoo.conf, others say that it is due to the socket configuration in the reverse proxy and others talk about not exposing the database list. Another option they also say is to use an ICE server which I assume must require an extra service.
> 
> What experience have you had with this and what configuration options using the copier have been successful for you?

I have noticed that in test mode the video call works well, however, in production mode it does not. Does anyone feel the same? I'm almost sure there must be something wrong in the configuration of the Traefik labels generated by the copier.

**Reply by pleirb** [reply]
Related case: #478

**Comment by pleirb** [comment]
This is the warning that appears when you try to use call or video call:

![imagen](https://github.com/Tecnativa/doodba-copier-template/assets/2092475/9960c013-2c52-4def-a1cb-b9cc392a1e29)

**Comment by pleirb** [comment]
@josep-tecnativa
@pedrobaeza 

Sorry to bother you but the intrigue is killing me. In development mode, video calls work perfectly. What is happening that in production mode this does not work? Is it a configuration thing in Traefik? I have searched everywhere on the internet and I can't find anything relevant that gives me at least a clue.

**Reply by pleirb** [reply]
I found some information that may be useful:
[https://github.com/odoo/odoo/issues/106339](https://github.com/odoo/odoo/issues/106339)

**Reply by pedrobaeza** [reply]
The websocket bindings are done here, so that shouldn't be. I confirm that video-calls work. The exclamation appears just at the beginning until the connection is properly established. Using ICE servers, the startup times will improve though.

**Reply by pleirb** [reply]
> The websocket bindings are done here, so that shouldn't be. I confirm that video-calls work. The exclamation appears just at the beginning until the connection is properly established. Using ICE servers, the startup times will improve though.

I don't understand what could be happening to me then. Either it's because I use subdomains or it's because the tests I'm doing with Traefik 3 are now being done in insecure mode (local tests by editing the hosts file)

**Reply by pedrobaeza** [reply]
Yes, with any of them.

**Reply by pleirb** [reply, marked as answer]
Well, I finally found the solution!
It turns out that in test.yaml this configuration for the odoo service is included:
```
    command:
      - odoo
      - --workers=2
      - --max-cron-threads=1
```
And in prod.yaml this is **NOT** included, I simply put the same thing and now everything works perfectly. @pedrobaeza Can you verify that this is included in your configuration? Because the template, at least I'm sure, does not generate these configurations for prod.yaml

The configuration I suppose can be changed according to needs. An example for the number of simultaneous video calls is provided in the odoo documentation in the LiveChat section:
[https://www.odoo.com/documentation/16.0/administration/on_premise/deploy.html](https://www.odoo.com/documentation/16.0/administration/on_premise/deploy.html)

**Reply by pedrobaeza** [reply]
We provide the proper configuration file in `odoo/custom/conf.d/`, including workers, and that's why we don't need that. The other one is explicitly put in the yaml file to override this configuration and don't steal too much resources to the machine.

**Reply by pleirb** [reply]
> We provide the proper configuration file in `odoo/custom/conf.d/`, including workers, and that's why we don't need that. The other one is explicitly put in the yaml file to override this configuration and don't steal too much resources to the machine.

I don't understand how resources are now being used only in test mode. But in prod mode by default this configuration is not provided to save resources by sacrificing the use of video calls?

**Reply by pedrobaeza** [reply]
We don't provide any configuration, because it totally depends on the hosted machine. Either 32 workers or 4 are incorrect depending on which server you run it.
